### DISCUSSION TITLE
Typescript: Typing Input Elements

### DISCUSSION THREAD

**Discussion by onebarloop** [neuefische/questions#424] [Web] [answered]
I have a typescript specific problem. 

In my upload component, I need to access the values of different named input field via `event.target.NAMEOFINPUTFIELD`

The code works fine, but typescript throws an error: "Die Eigenschaft "NAMEOFINPUTFIELD" ist für den Typ "EventTarget" nicht vorhanden."

```typescript
 async function onSubmit(event: React.SyntheticEvent) {
    event.preventDefault();

    const { artistname, firstname, lastname, location, pics } = event.target;

    const urls = await cloudUpload(pics.files);

    const newArtist = new Artist(
      artistname.value,
      firstname.value,
      lastname.value,
      location.value,
      urls
    );

    const responseDB = await fetch("/api", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(newArtist),
    });

    console.log(responseDB);
  }
```

I tried different approaches, with different types for the event, but none work. 

This is the [link](https://github.com/onebarloop/capstone-project/pull/20/files#diff-428cfa66537c17710aa7a7d184b3f51b00cbed8ddcf4f5457c0c5dcc8d5964bf) to the file in my PR 

Some hints would be much appreciated!

**Comment by actyralf** [comment, marked as answer]
I didn't test this but I guess you can prevent the error by extending the `React.SyntheticEvent` like this:

```js
const target = e.target as typeof e.target & {
      artistname: { value: string };
      firstname: { value: string };
      // etc.
    };
    const { artistname, firstname, lastname, location, pics } = target;
```
Please check this for further documentation: https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forms_and_events/

**Reply by onebarloop** [reply]
This absolutely works! I was pretty close, but didn't get the syntax right. Thanks a lot Ralf! 

Topic closed :)
